src/librbd/LibrbdWriteback.cc

```diff
@@ -163,6 +168,8 @@ namespace librbd {
 				 object_no, off, objectx, object_overlap,
 				 bl, snapc, snap_id,
 				 req_comp);
+    if (m_ictx->closing)
+      req->set_op_flags(LIBRADOS_OP_FLAG_FADVISE_DONTNEED);
```

Comment:
@dillaman  & @liewegas . Because DONTNEED is  dangerous and we don't easy suppose it(especially in rbd with multi user).  So i will remove this patch.
By the way, i wanted add cache option in cli rbd which let user to chose whether don't cache data. Only user know exactly. How about this?